frontend/src/filter_table.tsx

```diff
@@ -0,0 +1,218 @@
+  }
+});
+
+type Filter = {field: keyof(Individual), label: string, active?: boolean}
+
+/**
+ * Shows a table of individual information for the given list of `individuals`.
+ * Optionally `filters` or `title` can be set to further customize the table.
+ */
+export function FilterTable({individuals, title = '', filters = []}:
+    {individuals: Individual[], title: string, filters: Filter[]}) {
+
+  const { popup } = useMessageContext()
+  const styles = useStyles();
+
+  const defaultColumns = [
+    {field: 'herd', title: 'Besättning', hidden: false,
+      render: (rowData:any) =>
+        <a className={styles.functionLink}
+          onClick={() => popup(<HerdView id={rowData.herd['herd']} />, `/herd/${rowData.herd['herd']}`)}
+          >
+          {rowData.herd['herd']}
+        </a>
+    },
+    {field: 'name', title: 'Namn', hidden: false},
+    {field: 'certificate', title: 'Certifikat', hidden: false},
+    {field: 'number', title: 'Nummer', hidden: false,
+      render: (rowData:any) =>
+        <a className={styles.functionLink}
+          onClick={() => popup(<IndividualView id={rowData.number} />, `/individual/${rowData.number}`)}
+          >
+          {rowData.number}
+        </a>
+    },
+    {field: 'sex', title: 'Kön', hidden: false},
+    {field: 'birth_date', title: 'Födelsedatum', hidden: false},
+    {field: 'death_date', title: 'Dödsdatum', hidden: false},
+    {field: 'death_note', title: 'Dödsanteckning', hidden: false},
+    {field: 'mother', title: 'Moder', hidden: false,
+      render: (rowData:any) =>
+      <a className={styles.functionLink}
+        onClick={() => popup(<IndividualView id={rowData.mother['number']} />, `/individual/${rowData.mother['number']}`)}
+        >
+        {rowData.mother['number']}
+      </a>
+    },
+    {field: 'father', title: 'Fader', hidden: false,
+      render: (rowData:any) =>
+      <a className={styles.functionLink}
+        onClick={() => popup(<IndividualView id={rowData.father['number']} />, `/individual/${rowData.father['number']}`)}
+        >
+        {rowData.father['number']}
+      </a>
+    },
+    {field: 'color', title: 'Färg', hidden: false,
+      render: (rowData:any) => rowData.color['name']
+    },
+    {field: 'color_note', title: 'Färganteckning', hidden: false},
+  ]
+
+  const [columns, setColumns] = React.useState(defaultColumns)
+  const [currentFilters, setFilters] = React.useState(filters)
+
+  const updateColumns = (values: any[]) => {
+    const newColumns = [...columns];
+    const visibleColumns = values.map((v: any) => v.value);
+    newColumns.forEach((c: any) => {
+      if (visibleColumns.includes(c.field)) {
+        c.hidden = false;
+      } else {
+        c.hidden = true;
+      }
+    })
+    setColumns(newColumns)
+  }
+
+  const filteredIndividuals = () => {
+    if (!individuals) {
+      return []
+    }
+    return individuals.filter((i: Individual) => {
+      for (let f = 0; f < currentFilters.length; f++) {
```

Comment:
`for (let f of currentFilters)` if you want to type less :nerd_face:

Reply:
Good point :D I don't know why I always forget this syntax... I mean it's odd, but I'm supposed to know it xD

Reply:
I think the memory rule is 
- What would be sensible?
- `for ... in ...`
- Then what does JavaScript use?
- `for ... of ...`